realm/src/main/java/io/realm/internal/RealmCore.java

```diff
@@ -39,7 +38,7 @@
     private static final String JAVA_LIBRARY_PATH = "java.library.path";
 //*/
 
-    private static AtomicBoolean libraryIsLoaded = new AtomicBoolean(false);
+    private static boolean libraryIsLoaded = false;
```

Comment:
Either remove the Boolean or make it `static volatile` otherwise liveness may occur